deploy.go

```diff
@@ -464,14 +467,29 @@ func (d *imageDeployer) run(opts *deployOptions) error {
 
 	if fi.Mode()&os.ModeDevice != 0 {
 		// Source file is a block device
-		d.devPath = opts.Positional.Input
+		if err := d.initDevPathFormat(opts.Positional.Input); err != nil {
+			return err
+		}
 		return d.deployImageOnDevice()
 	}
 
 	// Source file is not a block device
 	return d.deployImageFromFile()
 }
 
+func (d *imageDeployer) checkPrerequisites() error {
+	if d.isNbdDevice() {
+		if !nbd.IsSupported() {
+			return errors.New("cannot create nbd devices (is qemu-nbd installed?)")
+		}
+		if !nbd.IsModuleLoaded() {
+			return errors.New("cannot create nbd devices because the required kernel module is not loaded")
+		}
+	}
+
+	return nil
+}
```

Comment:
It looks like this is called before `encryptCloudImageBase.devPath` is set so `encryptCloudImageBase.isNbdDevice` will always return false and the checks will never run.

Reply:
I think this should be taken care of now.